### ISSUE TITLE
Loadbalancer with Sticky sessions always forward the initial request to the same server

### ISSUE THREAD

**Issue by richierg** [containous/traefik#5521]
### Do you want to request a *feature* or report a *bug*?

Bug

### Did you try using a 1.7.x configuration for the version 2.0?

- [ ] Yes
- [x] No

### What did you do?

I'm trying to configure loadbalacing with traefik and sticky feature. (httpOnly and secure flag are enabled)

When I do my first request (without cookie in my browser). I'm alway redirect to the same server, despite the browser(Firefox,Chrome), private browsing or Laptop. All initial request are forwarded to the first server in the list (traefik dashboard).  

If I clean the cookie, and redo the request I can have another server.

### What did you expect to see?

The initial request need to be send to different server (RR) even from different browser or laptop.

### What did you see instead?

All initial request are forwarded to the first server in the list (traefik dashboard).

### Output of `traefik version`: (_What version of Traefik are you using?_)

```
Version:      2.0.1
Codename:     montdor
Go version:   go1.13.1
Built:        2019-09-26T16:18:03Z
OS/Arch:      linux/amd64

```

### What is your environment & configuration (arguments, toml, provider, platform, ...)?

```compose
version: "3.2"
services:
  app:
    image: containous/whoami
    deploy:
      replicas: 5
      labels:
        - traefik.enable=true
        - traefik.port=8080
        - traefik.http.routers.http-router.entrypoints=http
        - traefik.http.routers.http-router.rule=Host("whoami.xxx.com")
        - traefik.http.middlewares.http-to-https-redirect.redirectScheme.scheme=https
        - traefik.http.middlewares.http-to-https-redirect.redirectScheme.permanent=true
        - traefik.http.routers.http-router.middlewares=http-to-https-redirect
        - traefik.http.routers.https-router.entrypoints=https
        - traefik.http.routers.https-router.rule=Host("whoami.xxx.com")
        - traefik.http.services.whoami.loadbalancer.server.port=80
        - traefik.http.middlewares.hsts.headers.stsSeconds=315360000
        - traefik.http.routers.https-router.tls.options=myTLSOptions@file
        - traefik.http.routers.https-router.service=whoami
        - traefik.http.routers.https-router.middlewares=hsts
        - traefik.http.services.whoami.loadbalancer.sticky=true
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.name=traefik_xxx_cookie
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.httpOnly=true
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.secure=true
    networks:
      - traefik_net

  traefik:
    image:  traefik:v2.0
    command: |
        --accesslog.filePath="/var/log/traefik/access.log"
        --global.sendanonymoususage=false
        --log.level=debug
        --log.filePath="/var/log/traefik/traefik.log"
        --entrypoints.https.Address=:443
        --entrypoints.http.Address=:80
        --providers.docker
        --providers.docker.swarmmode=true
        --providers.file.filename="/certs/file_provider.yml"
        --providers.docker.exposedbydefault=false
        --api
        --api.insecure=true
    ports:
      - target: 80
        published: 80
        protocol: tcp
        mode: host
      - target: 443
        published: 443
        protocol: tcp
        mode: host
      - target: 8080
        published: 8080
        protocol: tcp
        mode: host
    deploy:
      replicas: 1
      placement:
        constraints: [node.role == manager]
    volumes:
      - type: bind
        source: "/var/run/docker.sock"
        target: "/var/run/docker.sock"
        read_only: true
      - type: bind
        source: "/srv/traefik/keys"
        target: "/certs"
      - type: bind
        source: "/srv/traefik/log"
        target: "/var/log/traefik"
    networks:
      - traefik_net

networks:
  traefik_net:
```

### If applicable, please paste the log output in DEBUG level (`--log.level=DEBUG` switch)

```
time="2019-09-27T10:10:55Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Cache-Control\":[\"max-age=0\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"none\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) snap Chromium/77.0.3865.90 Chrome/77.0.3865.90 Safari/537.36\"],\"X-Forwarded-Host\":[\"whoami.xxx.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"7319ffd8c193\"],\"X-Real-Ip\":[\"192.168.1.138\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"whoami.xxx.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.1.138:52436\",\"RequestURI\":\"/\",\"TLS\":null}"
time="2019-09-27T10:10:55Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Cache-Control\":[\"max-age=0\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"none\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) snap Chromium/77.0.3865.90 Chrome/77.0.3865.90 Safari/537.36\"],\"X-Forwarded-Host\":[\"whoami.xxx.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"7319ffd8c193\"],\"X-Real-Ip\":[\"192.168.1.138\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"whoami.xxx.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.1.138:52436\",\"RequestURI\":\"/\",\"TLS\":null}" ForwardURL="http://10.0.5.14:80"
-------------------------------------------------------------------------------------------------------------------------
time="2019-09-27T10:09:11Z" level=debug msg="vulcand/oxy/roundrobin/rr: begin ServeHttp on request" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"none\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36\"],\"X-Forwarded-Host\":[\"whoami.xxx.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"7319ffd8c193\"],\"X-Real-Ip\":[\"192.168.1.32\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"whoami.xxx.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.1.32:61559\",\"RequestURI\":\"/\",\"TLS\":null}"
time="2019-09-27T10:09:11Z" level=debug msg="vulcand/oxy/roundrobin/rr: Forwarding this request to URL" Request="{\"Method\":\"GET\",\"URL\":{\"Scheme\":\"\",\"Opaque\":\"\",\"User\":null,\"Host\":\"\",\"Path\":\"/\",\"RawPath\":\"\",\"ForceQuery\":false,\"RawQuery\":\"\",\"Fragment\":\"\"},\"Proto\":\"HTTP/2.0\",\"ProtoMajor\":2,\"ProtoMinor\":0,\"Header\":{\"Accept\":[\"text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8,application/signed-exchange;v=b3\"],\"Accept-Encoding\":[\"gzip, deflate, br\"],\"Accept-Language\":[\"fr-FR,fr;q=0.9,en-US;q=0.8,en;q=0.7\"],\"Sec-Fetch-Mode\":[\"navigate\"],\"Sec-Fetch-Site\":[\"none\"],\"Sec-Fetch-User\":[\"?1\"],\"Upgrade-Insecure-Requests\":[\"1\"],\"User-Agent\":[\"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/77.0.3865.90 Safari/537.36\"],\"X-Forwarded-Host\":[\"whoami.xxx.com\"],\"X-Forwarded-Port\":[\"443\"],\"X-Forwarded-Proto\":[\"https\"],\"X-Forwarded-Server\":[\"7319ffd8c193\"],\"X-Real-Ip\":[\"192.168.1.32\"]},\"ContentLength\":0,\"TransferEncoding\":null,\"Host\":\"whoami.xxx.com\",\"Form\":null,\"PostForm\":null,\"MultipartForm\":null,\"Trailer\":null,\"RemoteAddr\":\"192.168.1.32:61559\",\"RequestURI\":\"/\",\"TLS\":null}" ForwardURL="http://10.0.5.14:80"

```

**Comment by juliens** [comment]
Thanks for your interest in the project.

Can you try to use curl, multiple times and verify that it still contact the same whoami instance.

**Comment by richierg** [comment]
Thanks for the quick return.
I have the same issue with curl, the initial call is forwarded to the same server (as Firefox or an other curl on an other client). 
The others calls on the same clients change but all initials calls are forwarded to the same server (the first in servers list in traefik dashboard)

**Comment by richierg** [comment]
Hi @juliens 
Do you need more information ?

**Comment by richierg** [comment]
Any update please ? @juliens @dduportal 
The problem seems to be present in traefik 2.0.4

**Comment by dduportal** [comment]
Hi @richierg , this issue is labelled, and currently waiting for someone from the team finsihsing analyzing it. We have not forgotten it :)

**Comment by StargateSGx1** [comment]
hi,
i have the exact same behavior. I use also a sticky session (httpOnly=true, secure=false) and it directs me every time to the first server(from 6 Servers). Hope you can fix this bug.

**Comment by richierg** [comment]
Hi, the issue still present in release 2.1.1 :/ 
Do you have a lead ?

**Comment by richierg** [comment]
Hi @dduportal  , sorry to be insistent but I cannot upgrade my environment to Traefik 2.0 (2.1 now) because of this bug :/
Could you give me some information ?
Thanks

**Comment by dduportal** [comment]
@richierg as said before, it is a matter of someone from the team having the time and the opportunity to reproduce and study the behavior. If the issue is not updated, it is because this time was not found yet.

**Comment by guyromb** [comment]
Any update?

**Comment by mpl** [comment]
@guyromb I suspect there's a good chance that it might have been opportunistically fixed by https://github.com/containous/traefik/commit/98f304f8b0e6e89aee51d7d27a950dcf024ac4d0
So could you please retry with traefik v2.2 or master branch and let us know?

**Comment by guyromb** [comment]
It seems like TCP are not showing any longer in 2.2 @mpl

**Comment by guyromb** [comment]
@mpl I am still having the same issues. I actually disabled sticky session. And even without it same situation. Created an issue https://github.com/containous/traefik/issues/6566

**Comment by mpl** [comment]
@guyromb ah, but it looks like your TCP problems are irrelevant for this issue. The original poster was not using ingressroute or TCP as far as I can tell.

**Comment by mpl** [comment]
@richierg I could not use the configuration your provided because it does not work with my version of docker-compose, so I redid it with an equivalent simpler local configuration:

traefik.toml

```
[entryPoints]
  [entryPoints.web]
    address = ":80"

[api]
  insecure = true
  debug = true

[log]
  level = "DEBUG"
  format = "common"

[accesslog]

[providers]
  [providers.file]
    filename = "/Users/mpl/issues/5521/backends.toml"
```

backends.toml

```
[http.routers]
	[http.routers.simple]
		entryPoints = ["web"]
		service = "simple"
		rule = "Host(`simple.localhost`)"

[http.services]
	[http.services.simple]
		[http.services.simple.loadBalancer]
			[http.services.simple.loadBalancer.sticky.cookie]
				httpOnly = true
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9090"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9091"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9092"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9093"
```

With each server started as:

```
docker run -d -p 9090:80 containous/whoami
docker run -d -p 9091:80 containous/whoami
docker run -d -p 9092:80 containous/whoami
docker run -d -p 9093:80 containous/whoami
```

Then if I try to reproduce:

```
$ curl -vvv -H "Host:simple.localhost" http://localhost/whoami
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 80 (#0)
> GET /whoami HTTP/1.1
> Host:simple.localhost
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Length: 347
< Content-Type: text/plain; charset=utf-8
< Date: Wed, 25 Mar 2020 18:03:02 GMT
< Set-Cookie: _c2ee8=http://localhost:9090; Path=/; HttpOnly
<
Hostname: 5a4a6af23843
IP: 127.0.0.1
IP: 172.17.0.3
RemoteAddr: 172.17.0.1:57786
GET /whoami HTTP/1.1
Host: simple.localhost
User-Agent: curl/7.64.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: ::1
X-Forwarded-Host: simple.localhost
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: XXXXXXXXXX
X-Real-Ip: ::1
```
which shows the first server to be hit is `172.17.0.3`.
Then with a second attempt:

```
$ curl -vvv -H "Host:simple.localhost" http://localhost/whoami
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 80 (#0)
> GET /whoami HTTP/1.1
> Host:simple.localhost
> User-Agent: curl/7.64.1
> Accept: */*
>
< HTTP/1.1 200 OK
< Content-Length: 347
< Content-Type: text/plain; charset=utf-8
< Date: Wed, 25 Mar 2020 18:06:00 GMT
< Set-Cookie: _c2ee8=http://localhost:9091; Path=/; HttpOnly
<
Hostname: 133aa680a3e7
IP: 127.0.0.1
IP: 172.17.0.2
RemoteAddr: 172.17.0.1:53330
GET /whoami HTTP/1.1
Host: simple.localhost
User-Agent: curl/7.64.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: ::1
X-Forwarded-Host: simple.localhost
X-Forwarded-Port: 80
X-Forwarded-Proto: http
X-Forwarded-Server: XXXXXXXXX
X-Real-Ip: ::1
```

which shows that this time `172.17.0.2` is the one being hit.
So it seems to me that load-balancing of any first request works as expected in this case.

So either there is a fundamental difference between my config and yours that I missed, or we still do not understand the problem you think you observe.

What do you think?

**Comment by richierg** [comment]
Hi @mpl thanks for your feedback.
I try your solution without HTTPS and It's working (with the follow configuration )
```
version: "3.2"
services:
  app:
    image: containous/whoami
    deploy:
      replicas: 5
      labels:
        - traefik.enable=true
        - traefik.port=8080
        - traefik.http.routers.http-router.entrypoints=http
        - traefik.http.routers.http-router.rule=Host("lb3.localhost")
        - traefik.http.services.whoami.loadbalancer.server.port=80
        - traefik.http.services.whoami.loadbalancer.sticky=true
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.name=traefik_xxx_cookie
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.httpOnly=true
    networks:
      - traefik_net

  traefik:
    image:  traefik:v2.1
    command: |
        --accesslog.filePath="/var/log/traefik/access.log"
        --global.sendanonymoususage=false
        --log.level=debug
        --log.filePath="/var/log/traefik/traefik.log"
        --entrypoints.https.Address=:443
        --entrypoints.http.Address=:80
        --providers.docker
        --providers.docker.swarmmode=true
        --providers.file.filename="/certs/file_provider.yml"
        --providers.docker.exposedbydefault=false
        --api
        --api.insecure=true
    ports:
      - target: 80
        published: 80
        protocol: tcp
        mode: host
      - target: 443
        published: 443
        protocol: tcp
        mode: host
      - target: 8080
        published: 8080
        protocol: tcp
        mode: host
    deploy:
      replicas: 1
      placement:
        constraints: [node.role == manager]
    volumes:
      - type: bind
        source: "/var/run/docker.sock"
        target: "/var/run/docker.sock"
        read_only: true
      - type: bind
        source: "/srv/traefik/keys"
        target: "/certs"
      - type: bind
        source: "/srv/traefik/log"
        target: "/var/log/traefik"
    networks:
      - traefik_net

networks:
  traefik_net:

```
But when I set the TLS (as follow), the load balancer do the initial issue .. redirect on same host on first call despite the client :/ 
```
version: "3.2"
services:
  app:
    image: containous/whoami
    deploy:
      replicas: 5
      labels:
        - traefik.enable=true
        - traefik.port=8080
        - traefik.http.routers.https-router.entrypoints=https
        - traefik.http.routers.https-router.rule=Host("lb4.localhost")
        - traefik.http.services.whoami.loadbalancer.server.port=80
        - traefik.http.middlewares.hsts.headers.stsSeconds=315360000
        - traefik.http.routers.https-router.tls.options=myTLSOptions@file
        - traefik.http.routers.https-router.service=whoami
        - traefik.http.routers.https-router.middlewares=hsts
        - traefik.http.services.whoami.loadbalancer.sticky=true
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.name=traefik_xxx_cookie
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.httpOnly=true
        - traefik.http.services.whoami.loadbalancer.sticky.cookie.secure=true
    networks:
      - traefik_net

  traefik:
    image:  traefik:v2.1
    command: |
        --accesslog.filePath="/var/log/traefik/access.log"
        --global.sendanonymoususage=false
        --log.level=debug
        --log.filePath="/var/log/traefik/traefik.log"
        --entrypoints.https.Address=:443
        --entrypoints.http.Address=:80
        --providers.docker
        --providers.docker.swarmmode=true
        --providers.file.filename="/certs/file_provider.yml"
        --providers.docker.exposedbydefault=false
        --api
        --api.insecure=true
    ports:
      - target: 80
        published: 80
        protocol: tcp
        mode: host
      - target: 443
        published: 443
        protocol: tcp
        mode: host
      - target: 8080
        published: 8080
        protocol: tcp
        mode: host
    deploy:
      replicas: 1
      placement:
        constraints: [node.role == manager]
    volumes:
      - type: bind
        source: "/var/run/docker.sock"
        target: "/var/run/docker.sock"
        read_only: true
      - type: bind
        source: "/srv/traefik/keys"
        target: "/certs"
      - type: bind
        source: "/srv/traefik/log"
        target: "/var/log/traefik"
    networks:
      - traefik_net

networks:
  traefik_net:

```
The problem seems to be link with the SSL settings

**Comment by mpl** [comment]
No problem with TLS either here:

```
[entryPoints]
  [entryPoints.web]
    address = ":80"
  [entryPoints.secure]
    address = ":443"

[api]
  insecure = true
  debug = true

[log]
  level = "DEBUG"
  format = "common"

[accesslog]

[providers]
  [providers.file]
    filename = "/Users/mpl/issues/5521/backends.toml"
```

```
[http.routers]
	[http.routers.simple]
		entryPoints = ["web"]
		service = "simple"
		rule = "Host(`simple.localhost`)"
	[http.routers.simpletls]
		entryPoints = ["secure"]
		service = "simple"
		rule = "Host(`simpletls.localhost`)"
	[http.routers.simpletls.tls]

[http.services]
	[http.services.simple]
		[http.services.simple.loadBalancer]
			[http.services.simple.loadBalancer.sticky.cookie]
				httpOnly = true
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9090"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9091"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9092"
			[[http.services.simple.loadBalancer.servers]]
				url = "http://localhost:9093"
```

1st call:

```
$ curl -vvv -H "Host:simpletls.localhost" -k https://localhost/whoami
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=TRAEFIK DEFAULT CERT
*  start date: Mar 30 17:04:19 2020 GMT
*  expire date: Mar 30 17:04:19 2021 GMT
*  issuer: CN=TRAEFIK DEFAULT CERT
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7ffc5e802800)
> GET /whoami HTTP/2
> Host:simpletls.localhost
> User-Agent: curl/7.64.1
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200
< content-type: text/plain; charset=utf-8
< date: Mon, 30 Mar 2020 17:05:37 GMT
< set-cookie: _c2ee8=http://localhost:9090; Path=/; HttpOnly
< content-length: 355
<
Hostname: 5a4a6af23843
IP: 127.0.0.1
IP: 172.17.0.3
RemoteAddr: 172.17.0.1:57794
GET /whoami HTTP/1.1
Host: simpletls.localhost
User-Agent: curl/7.64.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: ::1
X-Forwarded-Host: simpletls.localhost
X-Forwarded-Port: 443
X-Forwarded-Proto: https
X-Forwarded-Server: XXXXXXXXXXX
X-Real-Ip: ::1

* Connection #0 to host localhost left intact
* Closing connection 0
```

2nd call:

```
curl -vvv -H "Host:simpletls.localhost" -k https://localhost/whoami
*   Trying ::1...
* TCP_NODELAY set
* Connected to localhost (::1) port 443 (#0)
* ALPN, offering h2
* ALPN, offering http/1.1
* successfully set certificate verify locations:
*   CAfile: /etc/ssl/cert.pem
  CApath: none
* TLSv1.2 (OUT), TLS handshake, Client hello (1):
* TLSv1.2 (IN), TLS handshake, Server hello (2):
* TLSv1.2 (IN), TLS handshake, Certificate (11):
* TLSv1.2 (IN), TLS handshake, Server key exchange (12):
* TLSv1.2 (IN), TLS handshake, Server finished (14):
* TLSv1.2 (OUT), TLS handshake, Client key exchange (16):
* TLSv1.2 (OUT), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (OUT), TLS handshake, Finished (20):
* TLSv1.2 (IN), TLS change cipher, Change cipher spec (1):
* TLSv1.2 (IN), TLS handshake, Finished (20):
* SSL connection using TLSv1.2 / ECDHE-RSA-AES256-GCM-SHA384
* ALPN, server accepted to use h2
* Server certificate:
*  subject: CN=TRAEFIK DEFAULT CERT
*  start date: Mar 30 17:04:19 2020 GMT
*  expire date: Mar 30 17:04:19 2021 GMT
*  issuer: CN=TRAEFIK DEFAULT CERT
*  SSL certificate verify result: unable to get local issuer certificate (20), continuing anyway.
* Using HTTP2, server supports multi-use
* Connection state changed (HTTP/2 confirmed)
* Copying HTTP/2 data in stream buffer to connection buffer after upgrade: len=0
* Using Stream ID: 1 (easy handle 0x7fa3fb005000)
> GET /whoami HTTP/2
> Host:simpletls.localhost
> User-Agent: curl/7.64.1
> Accept: */*
>
* Connection state changed (MAX_CONCURRENT_STREAMS == 250)!
< HTTP/2 200
< content-type: text/plain; charset=utf-8
< date: Mon, 30 Mar 2020 17:07:13 GMT
< set-cookie: _c2ee8=http://localhost:9091; Path=/; HttpOnly
< content-length: 355
<
Hostname: 133aa680a3e7
IP: 127.0.0.1
IP: 172.17.0.2
RemoteAddr: 172.17.0.1:53338
GET /whoami HTTP/1.1
Host: simpletls.localhost
User-Agent: curl/7.64.1
Accept: */*
Accept-Encoding: gzip
X-Forwarded-For: ::1
X-Forwarded-Host: simpletls.localhost
X-Forwarded-Port: 443
X-Forwarded-Proto: https
X-Forwarded-Server: XXXXXXXXXX
X-Real-Ip: ::1

* Connection #0 to host localhost left intact
* Closing connection 0
```

**Comment by richierg** [comment]
Hi @mpl , Could you try with docker stack and my compose file ?
Which version of docker compose do you use ? I can try to update my docker compose

**Comment by mpl** [comment]
@richierg I'm sorry, but unless you can show otherwise with a simple repro, it looks like Load-Balancing indeed works as expected even with Sticky set. So it looks like this might be a configuration issue, and we should move that discussion to the community forum.

**Comment by richierg** [comment]
Hi @mpl .
I have tried with your ssl solution and it's working. But I found the problem and I think it's a bug.
When I remove this label : 
- traefik.http.routers.https-router.middlewares=hsts of the docker compose on my previous post, the loadbalancer working well.
The problem is also present  with the redirect (  traefik.http.routers.http-router.middlewares=http-to-https-redirect) on my initial post. 
I think when we link a middleware to our router, the loadbalancer not work.
